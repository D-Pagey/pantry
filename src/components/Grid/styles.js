import styled from 'styled-components';

export const Wrapper = styled.ul`
    display: grid;
    grid-column-gap: 1rem;
    grid-template-columns: repeat(4, max-content);
    list-style: none;
    margin: 0;
    padding: 0;
    width: max-content;
`;

export const Heading = styled.h5`
    font-size: 18px;
    margin: 0 0 1rem;
`;
