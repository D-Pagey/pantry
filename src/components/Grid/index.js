import React, { Fragment, useContext } from 'react';
import { arrayOf, shape, string } from 'prop-types';
import { FirebaseContext } from '../ProviderFirebase';
import deleteIcon from './assets/delete.svg';
import * as S from './styles';

const Grid = ({ data }) => {
    const { updateFridge } = useContext(FirebaseContext);

    const handleDelete = (name) => () => {
        const filteredItems = data.filter((item) => item.name !== name);
        updateFridge(filteredItems);
    };

    return (
        <S.Wrapper>
            <S.Heading>Name</S.Heading>
            <S.Heading>Expires</S.Heading>
            <S.Heading>Quantity</S.Heading>
            <S.Heading>Amend</S.Heading>

            {data.map((item) => (
                <Fragment key={item.name}>
                    <li>{item.name}</li>
                    <li>{item.expires}</li>
                    <li>{item.servings} Servings</li>
                    <li>
                        {/* eslint-disable */}
                        <img
                            src={deleteIcon}
                            alt="delete"
                            onClick={handleDelete(item.name)}
                            style={{ cursor: 'pointer' }}
                        />
                    </li>
                </Fragment>
            ))}
        </S.Wrapper>
    );
};

Grid.propTypes = {
    data: arrayOf(
        shape({
            category: shape({
                label: string.isRequired,
                value: string.isRequired
            }).isRequired,
            expires: string.isRequired,
            name: string.isRequired,
            servings: string.isRequired
        }).isRequired
    ).isRequired
};

export default Grid;
