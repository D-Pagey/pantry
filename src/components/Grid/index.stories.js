import React from 'react';
import { storiesOf } from '@storybook/react';
import Grid from '.';

const props = {
    data: [
        {
            category: 'meat',
            name: 'chicken',
            servings: 2
        },
        {
            category: 'fish',
            name: 'salmon',
            servings: 1
        },
        {
            category: 'vegetables',
            name: 'carrots',
            servings: 3
        }
    ]
};

storiesOf('Grid', module).add('default', () => <Grid {...props} />);
