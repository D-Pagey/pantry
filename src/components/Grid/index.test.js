import React from 'react';
import dateFns from 'date-fns';
import Grid from '.';

const props = {
    data: [
        {
            category: { label: 'Meat', value: 'meat' },
            expires: dateFns.format(new Date(2019, 2, 14), 'MM/DD/YYYY'),
            name: 'chicken',
            servings: '2'
        },
        {
            category: { label: 'Fish', value: 'fish' },
            expires: dateFns.format(new Date(2019, 3, 9), 'MM/DD/YYYY'),
            name: 'salmon',
            servings: '1'
        },
        {
            category: { label: 'Vegetables', value: 'vegetables' },
            expires: dateFns.format(new Date(2014, 6, 11), 'MM/DD/YYYY'),
            name: 'carrots',
            servings: '3'
        }
    ]
};

describe('Grid component', () => {
    it('should render', () => {
        const { container } = render(<Grid {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
