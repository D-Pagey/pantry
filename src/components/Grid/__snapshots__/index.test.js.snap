// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render 1`] = `
.c0 {
  display: grid;
  grid-column-gap: 1rem;
  grid-template-columns: repeat(4,max-content);
  list-style: none;
  margin: 0;
  padding: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c1 {
  font-size: 18px;
  margin: 0 0 1rem;
}

<ul
  class="c0"
>
  <h5
    class="c1"
  >
    Name
  </h5>
  <h5
    class="c1"
  >
    Expires
  </h5>
  <h5
    class="c1"
  >
    Quantity
  </h5>
  <h5
    class="c1"
  >
    Amend
  </h5>
  <li>
    chicken
  </li>
  <li>
    03/14/2019
  </li>
  <li>
    2
     Servings
  </li>
  <li>
    <img
      alt="delete"
      src="delete.svg"
      style="cursor: pointer;"
    />
  </li>
  <li>
    salmon
  </li>
  <li>
    04/09/2019
  </li>
  <li>
    1
     Servings
  </li>
  <li>
    <img
      alt="delete"
      src="delete.svg"
      style="cursor: pointer;"
    />
  </li>
  <li>
    carrots
  </li>
  <li>
    07/11/2014
  </li>
  <li>
    3
     Servings
  </li>
  <li>
    <img
      alt="delete"
      src="delete.svg"
      style="cursor: pointer;"
    />
  </li>
</ul>
`;
